[DataGrid] Make GRID_ROWS_CLEAR private

oliviertassinari opened the pull request:
### Breaking changes

- [DataGrid] Make GRID_ROWS_CLEAR private (#1925) @oliviertassinari 
 The `GRID_ROWS_CLEAR` event was always triggered alongside `GRID_ROWS_SET`. You can listen to the latter event only.

---

~See the rationale in https://github.com/mui-org/material-ui-x/pull/1862#discussion_r654370166. When GRID_ROWS_CLEARED triggers, `GRID_ROWS_SET` is also triggered, no need to update the sorting state **twice**.~

~Note that this makes `GRID_ROWS_CLEARED` an orphan, nobody is listening to the event.~

~If I'm missing something and is really required, then it surfaces a hole in the test suite, the CI is green.~

dtassone commented:
I don't think it is dead code at least when I wrote it, it was not.
If I remember well it is to clear the sortedRows used sortedGridRowIdsSelector, to avoid getting an error when a row was added or removed and not in the lookup.

dtassone commented:
it seems ok now https://deploy-preview-1925--material-ui-x.netlify.app/storybook/?path=/story/x-grid-tests-streaming--slow-update-grid

dtassone commented on packages/grid/_modules_/grid/hooks/features/sorting/useGridSorting.ts:
```diff
@@ -3,7 +3,6 @@ import {
   GRID_COLUMN_HEADER_CLICK,
   GRID_COLUMN_HEADER_KEYDOWN,
   GRID_COLUMNS_UPDATED,
-  GRID_ROWS_CLEARED,
```
you can remove the constant as well and update the api

oliviertassinari replied:
This would be a breaking change. I think that we can decorate the two. My objective here is to simplify the logic.

dtassone replied:
we are in alpha. Let's do it I would say

dtassone commented:
> I don't think it is dead code at least when I wrote it, it was not.
> If I remember well it is to clear the sortedRows used sortedGridRowIdsSelector, to avoid getting an error when a row was added or removed and not in the lookup.

I think the rendering optimisation fixed it

m4theushw reviewed (commented):
This PR broke https://deploy-preview-1925--material-ui-x.netlify.app/storybook/?path=/story/x-grid-tests-filter--commodity-with-new-rows-via-api

The following diff passes on HEAD but here doesn't:

```diff
diff --git a/packages/grid/x-grid/src/tests/rows.XGrid.test.tsx b/packages/grid/x-grid/src/tests/rows.XGrid.test.tsx
index c800f9cb..591610b2 100644
--- a/packages/grid/x-grid/src/tests/rows.XGrid.test.tsx
+++ b/packages/grid/x-grid/src/tests/rows.XGrid.test.tsx
@@ -162,6 +162,7 @@ describe('<XGrid /> - Rows', () => {

     it('should allow to reset rows with setRows and render after 100ms', () => {
       render(<TestCase />);
+      expect(getColumnValues()).to.deep.equal(['Nike', 'Adidas', 'Puma']);
       const newRows = [
         {
           id: 3,
@@ -169,10 +170,9 @@ describe('<XGrid /> - Rows', () => {
         },
       ];
       apiRef.current.setRows(newRows);
-
-      clock.tick(50);
-      expect(getColumnValues()).to.deep.equal(['Nike', 'Adidas', 'Puma']);
-      clock.tick(50);
+      // Force an update before the 100ms
+      apiRef.current.forceUpdate(() => apiRef.current.state);
+      clock.tick(100);
       expect(getColumnValues()).to.deep.equal(['Asics']);
     });
```

My suggestion is to try to remove the timeout from the custom `forceUpdate` in `useGridRows` or to reset the sorted rows after updating the rows in this hook (might break the separation of concern).

oliviertassinari commented:
> This PR broke

@m4theushw Oh wow, great catch. I have added a failing test case to avoid regression in the future. I have also updated the assertion to illustrate the behavior. I have reduced the ambition of the changes.

dtassone commented:
Should we use this PR to rename the event to match the convention CLEARED => CLEAR?

oliviertassinari commented:
I'm waiting for #1862 to be merged to rebase.
